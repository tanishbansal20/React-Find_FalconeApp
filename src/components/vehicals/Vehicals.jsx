import React, {Component} from 'react';
import ReactRadioButtonGroup from 'react-radio-button-group';

export default class Vehicals extends Component {

  render() {
    const {destination, vehicalClass, vehicals, selectedVehical} = this.props.allProps;
    return (
      <div className={vehicalClass}>
        <ReactRadioButtonGroup
          name={vehicalClass}
          options={vehicals}
          value={selectedVehical}
          onChange={ (event) => this.props.vehicalSelectFun(event, destination)}
        />
      </div>
    )
  }
}
