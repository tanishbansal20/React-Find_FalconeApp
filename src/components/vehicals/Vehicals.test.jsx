import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import Vehicals from "./Vehicals.jsx";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

let propObject = {
  destination:"Destination1",
  vehicalClass:"vehicalRadioButton1",
  vehicals: [{value:"Space pot", label:"Space pot (2)"},{ value:"Rocket", label:"Rocket (1)"}],
  selectedVehical: "Rocket"
}

it("renders successfully with props", () => {

  act(() => {
    render(<Vehicals
      vehicalSelectFun={() => {}}
      allProps={propObject} />, container);
  });
  expect(container).toMatchSnapshot();
});
