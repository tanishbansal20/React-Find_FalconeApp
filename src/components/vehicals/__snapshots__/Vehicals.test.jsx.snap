// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully with props 1`] = `
<div>
  <div
    class="vehicalRadioButton1"
  >
    <div>
      <div>
        <input
          id="react-radio-button-group-1-Space pot"
          name="vehicalRadioButton1"
          type="radio"
          value="Space pot"
        />
        <label
          for="react-radio-button-group-1-Space pot"
        >
          Space pot (2)
        </label>
      </div>
      <div>
        <input
          checked=""
          id="react-radio-button-group-1-Rocket"
          name="vehicalRadioButton1"
          type="radio"
          value="Rocket"
        />
        <label
          for="react-radio-button-group-1-Rocket"
        >
          Rocket (1)
        </label>
      </div>
    </div>
  </div>
</div>
`;
